storage/src/adb/any.rs

```diff
@@ -386,8 +386,9 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> Any<E, K, V,
             .iter()
             .map(|op| Any::<E, _, _, _>::op_digest(hasher, op))
             .collect::<Vec<_>>();
+        let slices = digests.iter().map(|d| d.as_ref()).collect::<Vec<_>>();
 
-        proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
+        proof.verify_range_inclusion(hasher, &slices, start_pos, end_pos, root_hash)
```

Comment:
If you modify this to accept an iterator (like I did in #726), you can avoid these intermediate vectors.

Reply:
done

Reply:
This is the one spot where I still seem to need to collect() first since the operations need to be hashed into digests and rust complains if I try to take a reference to them as is expected by the verify_range_inclusion iterator.  HAven't figured out a way around this yet.

Reply:
With the change to AsRef<T> typed items we are now able to get rid of the collect() (thanks @danlaine ) but the trade-off is we need a temporary hasher. Not sure which is preferrable. I think I prefer the temp vector TBH since with generics you never really know the cost of the method invoked whereas temp vector cost is pretty well understood.